### DISCUSSION TITLE
Border routing on Android 15

### DISCUSSION THREAD

**Discussion by lmnotran** [openthread/openthread#10355] [Q&A]
Hi! 

I would like to test border routing functionality on a Google Pixel 8 running the latest Android 15 build and a EFR32 running `ot-rcp`. Is this something that's currently feasible?

Thanks!
Mason

**Comment by wgtdkp** [comment]
No, you won't be able to test with Pixel 8 because you need to change the Android code to add a new hardware to the existing phone. If you are only an Android device vendor and capable of building your OS image for your products, I can provide instructions to make EFR32 working as a RCP dongle plugged to your device.

**Reply by lmnotran** [reply]
@wgtdkp Thanks for responding! Assuming I am able to figure out how to build OS images, what would be the best hardware for me to use as an Android development platform if not a Pixel 8? Is it possible for me to pass the EFR32 USB device through to a Cuttlefish instance running on my PC? Sorry if these are dumb questions, I'm new to Android development

> I can provide instructions to make EFR32 working as a RCP dongle plugged to your device.

I would greatly appreciate that!

**Reply by wgtdkp** [reply]
> Is it possible for me to pass the EFR32 USB device through to a Cuttlefish instance running on my PC?

Yes this is feasible, @zhanglongxia did this before. And you will need code changes to the Thread HAL service and sepolicy rules.

Can you get your AOSP codebase ready and Thread BR in cuttlefish working? Then I can point you to where need to be updated.

**Reply by lmnotran** [reply]
@wgtdkp Thanks for the info! I'm looking into getting the hardware resources I need. I'll get back to you when I'm ready
